File: java/000/LeetCode0021.java
public class LeetCode0021{
    // 类似归并排序子过程，需要建立头结点
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode head = new ListNode(0);
        ListNode h0 = head;
        ListNode h1 = l1;
        ListNode h2 = l2;
        while (h1 != null && h2 != null){
            if (h1.val < h2.val){
                h0.next = h1;
                h1 = h1.next;
            } else {
                h0.next = h2;
                h2 = h2.next;
            }
            h0 = h0.next;
        }
        h0.next = h1 == null ? h2 : h1;
        return head.next;
    }
}

File: cpp/L000/LeetCode0021.cc
#include "../heads/list_node.h"

class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        ListNode head(0);
        ListNode *tail{&head};
        while (list1 != nullptr || list2 != nullptr) {
            if (list1 == nullptr) {
                tail->next = list2;
                list2 = list2->next;
            } else if (list2 == nullptr) {
                tail->next = list1;
                list1 = list1->next;
            } else if (list1->val <= list2->val) {
                tail->next = list1;
                list1 = list1->next;
            } else {
                tail->next = list2;
                list2 = list2->next;
            }
            tail = tail->next;
        }
        return head.next;
    }
};